### PR TITLE
dist/tools/cppcheck: fix all Cppcheck 1.82 errors

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

`cppcheck` errors are making merging #12558 more difficult so this PR fixes this.

### Testing procedure

- Run locally: `./dist/tools/cppcheck/check.sh` no errors should appear
- Green Murdock

Please take a closer look at 1e671ac and c2f593f,

1e671ac I'm not sure if `int sig = -1` is the best initial value for `sig`, although shouldn't really matter since its updated in the while loop..

c2f593f is because `RTT_FREQUENCY` doesn't have a default value so it results in a division by 0. Maybe there is a best way than duplicating the warning.

If preferred I can squash all commits fixing same errors.

### Issues/PRs references

#12558 can benefit from this.

### CODE REVIEW

**Comment by kaspar030** [comment]
Sorry, I commented in the commit view, @fjmolinas can you find the spots or should I move the comments?

File: sys/include/net/gnrc/gomach/gomach.h [hunk 1]
```diff
@@ -85,6 +85,9 @@ extern "C" {
 #define GNRC_GOMACH_SUPERFRAME_DURATION_US        (300LU * US_PER_MS)
 #endif
 
+#ifndef RTT_FREQUENCY
+#warning "RTT_FREQUENCY undefined."
```
Comment (kaspar030) [inline sys/include/net/gnrc/gomach/gomach.h]: I think we can make this an error. The define is clearly used in the division below.

File: cpu/native/irq_cpu.c [hunk 2]
```diff
@@ -229,7 +229,7 @@ int irq_is_in(void)
 int _native_popsig(void)
 {
     int nread, nleft, i;
-    int sig;
+    int sig = -1;
```
Comment (kaspar030) [inline cpu/native/irq_cpu.c]: "0" would be a better choice, as the only user (right below) uses this result as index into native_irq_handlers, which has a safe NULL as default in native_irq_handlers[0], but native_irq_handlers[-1] would segfault.

**Comment by kaspar030** [comment]
BTW, with "Cppcheck 1.89" on my arch system, I get a lot more [errors](https://gist.github.com/03a11f04f2f6b1fd8a6bfb271594c600).

**Comment by kaspar030** [comment]
Semicolon missing just before `nib_table.c:55`

**Comment by fjmolinas** [comment]
> Semicolon missing just before nib_table.c:55

Yep my bad, fixed.

File: cpu/native/irq_cpu.c (hunk 2 above)
Comment (fjmolinas) [reply cpu/native/irq_cpu.c]: ok!

File: sys/include/net/gnrc/gomach/gomach.h (hunk 1 above)
Comment (fjmolinas) [reply sys/include/net/gnrc/gomach/gomach.h]: done.

**Comment by fjmolinas** [comment]
> BTW, with "Cppcheck 1.89" on my arch system, I get a lot more errors.

Argh annoying, I'll address this later today, need to switch to another subject now.

**Comment by fjmolinas** [comment]
> > BTW, with "Cppcheck 1.89" on my arch system, I get a lot more errors.
> 
> Argh annoying, I'll address this later today, need to switch to another subject now.

Or deal with it later on, I think it would be good to get rid of #12558. I could open an issue and deal with it later.

**Comment by kaspar030** [comment]
> Or deal with it later on, I think it would be good to get rid of #12558. I could open an issue and deal with it later.

sure! small steps are good. please squash!

**Comment by fjmolinas** [comment]
Opened #12771, I tackle it as soon as possible.

**Comment by fjmolinas** [comment]
@kaspar030 can I hit the button?

**Comment by kaspar030** [comment]
Thanks for dealing with this!

**Pull request merged**